query/expression.go

```diff
@@ -34,7 +34,8 @@ type EvaluationContext struct {
 	API          api.API          // Api to obtain metadata from
 	Timerange    api.Timerange    // Timerange to fetch data from
 	SampleMethod api.SampleMethod // SampleMethod to use when up/downsampling to match the requested resolution
-	Predicate    api.Predicate
+	Predicate    api.Predicate    // Predicate to apply to TagSets prior to fetching
+	FetchLimit   *int             // A limit on the number of fetches which may be performed
```

Comment:
yes, we (currently) want to share a single value amongst the entire execution, but we're copying the struct around thus we need to have an integer pointer.

we may need to make it into an atomic integer to avoid race conditions. Right now, we don't implement it, but i can see implementing parallel fetches amongst subexpressions. i.e. if you have 

```
select cpu_user + cpu_kernel where ...
```
you can fetch the left-hand-side and right-hand-side concurrently. Right now, our parallel fetching logic only fetches in-parallel within a single fetch expression.